### ISSUE TITLE
Research load testing

### ISSUE THREAD

**Issue by katbow** [healthlocker/healthlocker#804]
Need to look into the general concept of load/performance testing (@naazy @roryc89 would love your advice/knowledge on this!), and specifically with a phoenix application.

Spend ~1hour tomorrow morning for research purposes investigating how to test, what to test, any specifications for a phoenix app, etc. Come up with a plan for how to do this and verify.

**Comment by katbow** [comment]
## General concept
Load testing is an important part of the development process when your app is nearing "completion". Even though apps will have gone through testing throughout development, the only way to know with certainty that the application can handle the demands users will be putting on it is with load testing. This allows you to put simulated demand on the app so its behaviour can be checked under different conditions. 

Load testing can help answer questions and identify issues such as:
* Is the app doing what I expect under *X* conditions?
* Are there page load issues?
* What is the app's breaking point?

There are some [open source tools](https://www.blazemeter.com/blog/open-source-load-testing-tools-which-one-should-you-use) for load testing. However, this can also be done by having tests which add a lot of data e.g. a lot of users creating a lot of content, and checks the response time of various endpoints. [Page load time](https://stackoverflow.com/questions/40507927/how-to-measure-page-load-time-in-phoenix) can be accessed in phoenix from [Plug](https://hexdocs.pm/plug/readme.html).

One other little related tidbit I found to improve load times is to [change the log to `:warn`](https://elixirforum.com/t/evaluating-elixir-phoenix-for-a-web-scale-performance-critical-application/832/5) in production.

TL;DR Plan is to create tests which will run as part of CI that check how long it takes for pages to load when a lot of content is generated. @nelsonic do you think it would be good to also look more into using an open source load testing tool?

**Comment by nelsonic** [comment]
@katbow try: https://github.com/shoreditch-ops/artillery 🚀 
@hassy is an awesome guy and will help if you have questions and ask nicely ... ;-)

**Comment by markwilliamfirth** [comment]
@katbow can this issue be closed?

**Comment by katbow** [comment]
yes 🙂
